### PR TITLE
remove unittest test_models

### PR DESCRIPTION
**Pull request by cl0cks4fe**
- removes unittest from `openpilot.selfdrive.car.tests.test_models.py`
- switches `tools/car_porting/test_car_model.py` to inherit from `test_models.py`

not totally ideal but keeps use

### CODE REVIEW

**Comment by cl0cks4fe** [comment]
skips base class tests but so does `test_models`

File: tools/car_porting/test_car_model.py [hunk 1]
```diff
@@ -1,38 +1,20 @@
...
-    # create new test case and discover tests
-    test_case_args = {"platform": test_route.car_model, "test_route": test_route, "test_route_on_bucket": ci}
-    CarModelTestCase = type("CarModelTestCase", (TestCarModel,), test_case_args)
-    test_suite.addTest(unittest.TestLoader().loadTestsFromTestCase(CarModelTestCase))
-  return test_suite
+assert route is not None, "Must specify route or segment name"
 
+route_or_segment_name = SegmentName(route, allow_route_name=True)
+segment_num = route_or_segment_name.segment_num if route_or_segment_name.segment_num != -1 else None
+test_route = CarTestRoute(route_or_segment_name.route_name.canonical_name, car, segment=segment_num)
```
Comment (adeebshihadeh) [inline tools/car_porting/test_car_model.py:16]: all this stuff runs at import time now. it should only be run if we're running this script

File: tools/car_porting/test_car_model.py [hunk 2]
```diff
@@ -1,38 +1,20 @@
...
-import unittest # noqa: TID251
+import os
+from typing import Any
 
 from openpilot.selfdrive.car.tests.routes import CarTestRoute
-from openpilot.selfdrive.car.tests.test_models import TestCarModel
+from openpilot.selfdrive.car.tests.test_models import TestCarModelBase
 from openpilot.tools.lib.route import SegmentName
 
+route = os.getenv('ROUTE')
```
Comment (adeebshihadeh) [inline tools/car_porting/test_car_model.py:9]: this is now harder to use without the argparser. can we bring that back?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

**Comment by cl0cks4fe** [comment]
@adeebshihadeh because of pytest discovery, lose something similar to `loadTestsFromTestCase` to run this as script, i'll revisit

**Pull request closed without merging**